File: Java/p11417.java

import java.util.Scanner;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author JASONYEHEZKIEL
 */
public class p11417 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        while (true) {
            int n = input.nextInt();
            if (n == 0)break;
            int res = 0;
            for (int i = 1; i < n; i++) {
                for (int j = i+1; j <= n; j++) {
                    res += findGCD(i,j);
                }
            }
            
            System.out.println(res);
        }
    }
    
    static int findGCD(int num1, int num2) {
        while (num1 != 0 && num2 != 0) {
            int c = num1%num2;
            num1 = num2;
            num2 = c;
        }
        
        if (num1 == 0) return num2;
        else return num1;
    }
}

File: Python/p11417.py
def findGCD(num1, num2):
    while num1 != 0 and num2 != 0:
        c = num1%num2
        num1 = num2
        num2 = c
        
    if num1 == 0:
        return num2
    else :
        return num1

while (True):
    n = int(input())
    if n == 0:
        break
    
    g = 0
    for i in range(1,n):
        for j in range(i+1, n+1):
            g += findGCD(i,j)
            
    print(g)